### PR TITLE
110 recreate connection of landing page maps and endpoint

### PR DESCRIPTION
**Pull request by ywkim312**
## Description
To test this, please set your local environment variable APP_ENV=development by
export APP_ENV=development
or
export APP_ENV=local

If you use local, you must run your api of the following PR to see the correct values
https://github.com/policy-design-lab/pdl-api/pull/66

If you use development, the values in the tab might not be correct, because the values in the database is not correct, currently. Also, in the SNAP map, if you hover the mouse, the Avg. Monthly Participation will be NaN. Because the same reason as above. 

To see the correct value, you have to set APP_ENV=local, run the above API PR, then connect VPN to ncsa. Then it will give you all the correct values.

However, if you see all the maps on landing page and the tabs for the values without any problem, in any APP_ENV, this PR works.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation and CHANGELOG accordingly.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

**Comment by pengyin-shan** [comment]
Tested using development API and the map appears _without any problem._ Did see the data difference you mentioned in PR.

Tested again using local API with `65-make-api-to-serve-summary-and-all-programs-directly-from-json` branch in pdl-api:
Screenshot 2023-05-16 at 1 43 26 PM
Title I, Title II and SNAP don't work, as explained by the CORS warning in the console when running local API:
Screenshot 2023-05-16 at 1 47 16 PM

Based on the dev api testing result, your PR should work. I wonder if it would be easier if you merge the pdl-api pr, then verify again with dev api directly?

**Comment by ywkim312** [comment]
> Tested using development API and the map appears _without any problem._ Did see the data difference you mentioned in PR.
> 
> Tested again using local API with `65-make-api-to-serve-summary-and-all-programs-directly-from-json` branch in pdl-api: Screenshot 2023-05-16 at 1 43 26 PM Title I, Title II and SNAP don't work, as explained by the CORS warning in the console when running local API: Screenshot 2023-05-16 at 1 47 16 PM
> 
> Based on the dev api testing result, your PR should work. I wonder if it would be easier if you merge the pdl-api pr, then verify again with dev api directly?

merging API will only change the values to correct values and without merging api PR, you can still test to see if the map works. However, merging API is also good idea to check if the values are rendered correctly.  It is kind of weird that I was able to test with my local env without cors error. Let me test it later and will let you know again

**Comment by ywkim312** [comment]
> Tested using development API and the map appears _without any problem._ Did see the data difference you mentioned in PR.
> 
> Tested again using local API with `65-make-api-to-serve-summary-and-all-programs-directly-from-json` branch in pdl-api: Screenshot 2023-05-16 at 1 43 26 PM Title I, Title II and SNAP don't work, as explained by the CORS warning in the console when running local API: Screenshot 2023-05-16 at 1 47 16 PM
> 
> Based on the dev api testing result, your PR should work. I wonder if it would be easier if you merge the pdl-api pr, then verify again with dev api directly?

@pengyin-shan i think the cors comes from the database not frontend. Can you try again with vpn connected to ncsa?

**Comment by pengyin-shan** [comment]
> > Tested using development API and the map appears _without any problem._ Did see the data difference you mentioned in PR.
> > Tested again using local API with `65-make-api-to-serve-summary-and-all-programs-directly-from-json` branch in pdl-api: Screenshot 2023-05-16 at 1 43 26 PM Title I, Title II and SNAP don't work, as explained by the CORS warning in the console when running local API: Screenshot 2023-05-16 at 1 47 16 PM
> > Based on the dev api testing result, your PR should work. I wonder if it would be easier if you merge the pdl-api pr, then verify again with dev api directly?
> 
> @pengyin-shan i think the cors comes from the database not frontend. Can you try again with vpn connected to ncsa?

Emm, just tried vpn and still see the CORS error, but probably I just miss some settings...I think your PR works since I tested without any problem using dev api. If Sandeep or Chris tested and it worked, you can merge this PR.

**Comment by sandeep-ps** [comment]
I was able to connect this branch with the web service PR branch running locally. For the most part, things look fine. It might still be good to merge the web service PR first and test it with develop branch (as some changes from develop need to be merged in that PR) before merging this one.

**Comment by ywkim312** [comment]
> I was able to connect this branch with the web service PR branch running locally. For the most part, things look fine. It might still be good to merge the web service PR first and test it with develop branch (as some changes from develop need to be merged in that PR) before merging this one.

sounds good to me. Let me know when the api one merged then I will deploy it to dev server.

**Comment by sandeep-ps** [comment]
I have merged the web service PR.

File: CHANGELOG.md [hunk 1]
```diff
@@ -13,6 +13,7 @@ and this project adheres to [Semantic Versioning](http://semver.org/spec/v2.0.0.
 
 ### Changed
 - eslint rules back to original [#94](https://github.com/policy-design-lab/pdl-frontend/issues/94)
+- Landing page maps and tabs uses api enpoint instead of local json file [#110](https://github.com/policy-design-lab/pdl-frontend/issues/110)
```
Comment (sandeep-ps) [inline CHANGELOG.md:16]: Please move this entry to the unreleased section. Thanks.

**Review by sandeep-ps — APPROVED** [review]
I tested this PR by running modules locally and comparing them with the development and production instances. All changes look good to me. Approving with a comment on updating the CHANGELOG. Thanks.

**Comment by pengyin-shan** [comment]
Switched to dev api and re-tested this PR. Everything looks good except the Avg. Monthly Participation for SNAP become 'NaN':
Screenshot 2023-05-16 at 4 46 02 PM

Is any column name or data format change on the SNAP side?

**Comment by sandeep-ps** [comment]
I can see monthly participation when everything runs locally: 
image
 Could it be a cache issue? Could you try after clearing the cache or from an incognito window?

**Comment by ywkim312** [comment]
I had the same problem when I tested. I believe that it is cache thing. In my case, it worked in incognito and it worked eventually after several refreshes.

**Comment by pengyin-shan** [comment]
This is what tried again under NCSA VPN:
1. using dev api (APP_ENV=development) has the NaN issue (try again in incognito window with cleared cache)
2. using the newest development branch of pdl-api (APP_ENV=local) still gives me the CORS error...

Awkward...I assume if I use `APP_ENV=development`, then app will query the dev endpoint right? We just merged a PR in pdl-api as I can remember, so am I supposed to see new data points after changing to `APP_ENV=development`?

Switched to `APP_ENV=development` and this is the data source: 
Screenshot 2023-05-16 at 5 13 28 PM


Anyway, pls feel free to go ahead if you feel it works. The problem could come from my side.

File: CHANGELOG.md (hunk 1 above)
Comment (ywkim312) [reply CHANGELOG.md]: changed.

**Comment by ywkim312** [comment]
@pengyin-shan Hi Wendy, could you please try one more time? The Nan problem should not be there anymore. The dev api has been updated by merging the PR to develop. You don't need VPN to be connected. Just simply use APP_ENV=development

**Comment by pengyin-shan** [comment]
> @pengyin-shan Hi Wendy, could you please try one more time? The Nan problem should not be there anymore. The dev api has been updated by merging the PR to develop. You don't need VPN to be connected. Just simply use APP_ENV=development

Sure, I just switched to development and the NaN issue is gone. This is the data points:

Screenshot 2023-05-19 at 2 45 39 PM
Is this what I am supposed to get?

**Comment by ywkim312** [comment]
> > @pengyin-shan Hi Wendy, could you please try one more time? The Nan problem should not be there anymore. The dev api has been updated by merging the PR to develop. You don't need VPN to be connected. Just simply use APP_ENV=development
> 
> Sure, I just switched to development and the NaN issue is gone. This is the data points:
> 
> Screenshot 2023-05-19 at 2 45 39 PM Is this what I am supposed to get?

It looks like it. However, as for the value comparison, could you compare with values in the prod? They should be the same

**Comment by pengyin-shan** [comment]
> > > @pengyin-shan Hi Wendy, could you please try one more time? The Nan problem should not be there anymore. The dev api has been updated by merging the PR to develop. You don't need VPN to be connected. Just simply use APP_ENV=development
> > 
> > 
> > Sure, I just switched to development and the NaN issue is gone. This is the data points:
> > Screenshot 2023-05-19 at 2 45 39 PM Is this what I am supposed to get?
> 
> It looks like it. However, as for the value comparison, could you compare with values in the prod? They should be the same

Yes, they are the same prod, as far as I can see.

**Review by pengyin-shan — APPROVED** [review]
Works well, thank you!

**Pull request merged**